File: rmt/src/test/scala/de/flower/rmt/test/WicketTests.scala
package de.flower.rmt.test

import org.springframework.context.ApplicationContext
import org.apache.wicket.protocol.http.WebApplication
import de.flower.rmt.ui.app.TestApplication
import org.testng.annotations.{AfterMethod, BeforeMethod}
import org.mockito.Mockito._

/**
 * 
 * @author flowerrrr
 */

class WicketTests extends AbstractIntegrationTests {


    protected def createWebApp(ctx: ApplicationContext, flag: Boolean): WebApplication = {
        return new TestApplication(ctx);
    }

    @BeforeMethod def init: Unit = {
        createTester(applicationContext)
    }

    @AfterMethod def cleanup: Unit = {
        validateMockitoUsage
    }

    private def createTester(ctx: ApplicationContext): Unit = {
        val webApp: WebApplication = createWebApp(ctx, true)
        wicketTester = new RMTWicketTester(webApp)
    }



    protected var wicketTester: RMTWicketTester = null

}

File: rmt/src/test/java/de/flower/rmt/test/WicketTests.java
package de.flower.rmt.test;

import de.flower.rmt.ui.app.TestApplication;
import org.apache.wicket.protocol.http.WebApplication;
import org.mockito.Mockito;
import org.springframework.context.ApplicationContext;
import org.testng.annotations.AfterMethod;
import org.testng.annotations.BeforeMethod;

/**
 * @author flowerrrr
 */

public class WicketTests extends AbstractIntegrationTests {

    protected RMTWicketTester wicketTester = null;

    protected WebApplication createWebApp(ApplicationContext ctx) {
        return new TestApplication(ctx);
    }

    @BeforeMethod
    public void init() {
        createTester(applicationContext);
    }

    @AfterMethod
    public void cleanup() {
        Mockito.validateMockitoUsage();
    }

    private void createTester(ApplicationContext ctx) {
        WebApplication webApp = createWebApp(ctx);
        wicketTester = new RMTWicketTester(webApp);
    }
}